### PR TITLE
Fix configuration files

### PR DESCRIPTION
**Pull request by szepeviktor**
@sybrew When started working I've found the config files need some care.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -39,5 +39,11 @@
...
 	},
 	"minimum-stability": "dev",
-	"prefer-stable": true
+	"prefer-stable": true,
+	"config": {
+		"allow-plugins": {
+			"composer/installers": true,
+			"dealerdirect/phpcodesniffer-composer-installer": true
+		}
+	}
```
Comment (szepeviktor) [inline composer.json:48]: Upgrade to Composer v2.
It is almost 3 years old.
https://github.com/composer/composer/releases/tag/2.0.0

File: .editorconfig [hunk 2]
```diff
@@ -4,4 +4,6 @@ root = true
 indent_style = tab
 indent_size = 4
 end_of_line = lf
+charset = utf-8
+trim_trailing_whitespace = true
```
Comment (szepeviktor) [inline .editorconfig:8]: These two keys were missing.
https://spec.editorconfig.org/#supported-pairs

File: wpml-config.xml [hunk 3]
```diff
@@ -1,13 +1,13 @@
...
+		<custom-field action="translate">_genesis_title</custom-field>
+		<custom-field action="translate">_genesis_description</custom-field>
+		<custom-field action="translate">_social_image_url</custom-field>
+		<custom-field action="translate">_genesis_canonical_uri</custom-field>
+		<custom-field action="translate">redirect</custom-field>
+		<custom-field action="translate">_open_graph_title</custom-field>
+		<custom-field action="translate">_twitter_title</custom-field>
+		<custom-field action="translate">_open_graph_description</custom-field>
+		<custom-field action="translate">_twitter_description</custom-field>
+	</custom-fields>
```
Comment (szepeviktor) [inline wpml-config.xml:12]: Not obeying editorconfig.

File: .gitignore [hunk 4]
```diff
@@ -22,3 +22,7 @@
 .Trashes
 ehthumbs.db
 Thumbs.db
+
+## Packages
+/composer.lock
+/vendor/
```
Comment (szepeviktor) [inline .gitignore:28]: Ignore Composer lock file and dependencies.

**Comment by szepeviktor** [comment]
BTW gitignore has - at least - [6 categories](https://github.com/szepeviktor/debian-server-tools/blob/45815d59350ea6cfd93617e3a589ccc942182aaf/.gitignore#L6-L12).
You feel much better when ignoring "OS" and "IDE and local environment" files globally in your user home
`git config --global core.excludesFile ~/.gitignore`
(as git's docs suggest https://git-scm.com/docs/gitignore)

File: composer.json (hunk 1 above)
Comment (sybrew) [reply composer.json:48]: I don't use Composer for WordPress, only standalone stuff (like WPCS) -- this file is mostly set up by those who do use it inside WordPress, e.g., by using [Bedrock](https://roots.io/bedrock/), and I rely on their feedback when an upgrade is required 😅. 

I don't know what the config/allow-plugins would limit, but what you added seems standard.

**Comment by sybrew** [comment]
> BTW gitignore has - at least - [6 categories](https://github.com/szepeviktor/debian-server-tools/blob/45815d59350ea6cfd93617e3a589ccc942182aaf/.gitignore#L6-L12). You will feel much better when ignoring "OS" and "IDE and local environment" files globally in your user home (not in every project) `git config --global core.excludesFile ~/.gitignore` (as git's docs suggest https://git-scm.com/docs/gitignore)

I didn't know it had a standard for categories; I just followed what seemed most logical and blocked files I stumbled upon across my systems 😅.

Feel free to add a commit that blocks all the OS junk, and although I'll never use Linux for writing code, I'm sure others might.

**Comment by szepeviktor** [comment]
> I didn't know it had a standard for categories

These are viktor-invented categories 😲

> I'll never use Linux for writing code

Windows has a user home also.
`git config --global core.excludesFile %USERPROFILE%/.gitignore`

> Feel free to add a commit that blocks all the OS junk

In the repo it needs to be deleted, that is the clue! Not to carry those in each and every project.
On your machine it needs to be ignored.

File: composer.json (hunk 1 above)
Comment (szepeviktor) [reply composer.json:48]: You have met your man.
Please see https://github.com/szepeviktor/composer-managed-wordpress

**Pull request merged**